### PR TITLE
A new and cleaner implementation of the symplectic Runge-Kutta-Nyström integrator

### PR DESCRIPTION
**Pull request by eggrobin**

### CODE REVIEW

File: integrators/symplectic_runge_kutta_nyström_integrator.hpp [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+class SymplecticRungeKuttaNyströmIntegrator
+    : public FixedStepSizeIntegrator<
+                 SpecialSecondOrderDifferentialEquation<Position>> {
+  static int const stages_ = composition == kBA ? evaluations : evaluations + 1;
+ public:
+  SymplecticRungeKuttaNyströmIntegrator(FixedVector<double, stages_> const& a,
+                                        FixedVector<double, stages_> const& b);
+
+  void Solve(IntegrationProblem<ODE> const& problem,
+             Time const& step) const override;
```
Comment (pleroy) [inline integrators/symplectic_runge_kutta_nyström_integrator.hpp]: Blank line after.

File: integrators/symplectic_runge_kutta_nyström_integrator_body.hpp [hunk 2]
```diff
@@ -0,0 +1,336 @@
...
+  }
+  CHECK_EQ(1.0, c_i.value);
+  if (composition == kABA) {
+    CHECK_EQ(0.0, b_[0]);
+  } else if (composition == kBAB) {
+    CHECK_EQ(0.0, a_[stages_ - 1]);
+  }
+  if (time_reversible) {
+    switch (composition) {
+    case kABA:
```
Comment (pleroy) [inline integrators/symplectic_runge_kutta_nyström_integrator_body.hpp]: All the case and their statements indent += 2.

File: integrators/symplectic_runge_kutta_nyström_integrator_body.hpp [hunk 3]
```diff
@@ -0,0 +1,336 @@
...
+
+  // Current Runge-Kutta-Nyström stage.
+  std::vector<Position> q_stage(dimension);
+  // Accelerations at the current stage.
+  std::vector<Acceleration> g(dimension);
+
+  bool at_end = false;
+
+  for (;;) {
+    // TODO(egg): do we want to keep cover the whole integration interval, or
```
Comment (pleroy) [inline integrators/symplectic_runge_kutta_nyström_integrator_body.hpp]: "keep cover"?

File: integrators/symplectic_runge_kutta_nyström_integrator_body.hpp [hunk 4]
```diff
@@ -0,0 +1,336 @@
...
+    // Increment the solution.
+    t.Increment(h);
+    for (int k = 0; k < dimension; ++k) {
+      q[k].Increment(∆q[k]);
+      v[k].Increment(∆v[k]);
+    }
+    problem.append_state(current_state);
+  }
+}
+
```
Comment (pleroy) [inline integrators/symplectic_runge_kutta_nyström_integrator_body.hpp]: Why so many blank lines?

File: integrators/symplectic_runge_kutta_nyström_integrator.hpp (hunk 1 above)
Comment (eggrobin) [inline integrators/symplectic_runge_kutta_nyström_integrator.hpp]: Done.

File: integrators/symplectic_runge_kutta_nyström_integrator_body.hpp (hunk 2 above)
Comment (eggrobin) [inline integrators/symplectic_runge_kutta_nyström_integrator_body.hpp]: Done.

File: integrators/symplectic_runge_kutta_nyström_integrator_body.hpp (hunk 3 above)
Comment (eggrobin) [inline integrators/symplectic_runge_kutta_nyström_integrator_body.hpp]: We discussed that comment and came to the conclusion that the currently-implemented behaviour was fine anyway.

File: integrators/symplectic_runge_kutta_nyström_integrator_body.hpp (hunk 4 above)
Comment (eggrobin) [inline integrators/symplectic_runge_kutta_nyström_integrator_body.hpp]: Don't know, maybe copy-paste? removing.

**Pull request merged**